### PR TITLE
Refactored ILambdaExpression to IAnonymousFunctionExpression, Signatu…

### PR DESCRIPTION
**Pull request by 333fred**
…re to Symbol. Fixes https://github.com/dotnet/roslyn/issues/21286. Tagging @dotnet/analyzer-ioperation @dotnet/roslyn-compiler for review.

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 1]
```diff
@@ -331,13 +331,13 @@ Namespace Microsoft.CodeAnalysis.Semantics
             Return New LazyNameOfExpression(argument, _semanticModel, syntax, type, constantValue)
         End Function
 
-        Private Function CreateBoundLambdaOperation(boundLambda As BoundLambda) As ILambdaExpression
-            Dim signature As IMethodSymbol = boundLambda.LambdaSymbol
+        Private Function CreateBoundLambdaOperation(boundLambda As BoundLambda) As IAnonymousFunctionExpression
+            Dim symbol As IMethodSymbol = boundLambda.LambdaSymbol
             Dim body As Lazy(Of IBlockStatement) = New Lazy(Of IBlockStatement)(Function() DirectCast(Create(boundLambda.Body), IBlockStatement))
             Dim syntax As SyntaxNode = boundLambda.Syntax
             Dim type As ITypeSymbol = boundLambda.Type
```
Comment (genlu) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: Unrelated to this change: can we put an assert here and simply set type to `Nothing`?

**Review by genlu — APPROVED** [review]

**Review by cston — APPROVED** [review]

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by agocke** [comment]
Why?

**Comment by 333fred** [comment]
@agocke the design committee decided to rename this because it's not just used for lambda expressions, but for all anonymous functions such as anonymous delegates.

**Comment by 333fred** [comment]
@dotnet-bot retest windows_debug_unit64_prtest please

File: src/Compilers/Core/Portable/Operations/IAnonymousFunctionExpression.cs [hunk 2]
```diff
@@ -5,20 +5,20 @@
 namespace Microsoft.CodeAnalysis.Semantics
 {
     /// <summary>
-    /// Represents a lambda expression.
+    /// Represents a anonymous function expression, such as a lambda or anonymous delegate.
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/Operations/IAnonymousFunctionExpression.cs]: >a anonymous [](start = 19, length = 11)

"**a** anonymous" vs. "**an** anonymous"

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 3]
```diff
@@ -286,7 +286,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
         End Function
 
         Private Function CreateBoundMeReferenceOperation(boundMeReference As BoundMeReference) As IInstanceReferenceExpression
-            Dim instanceReferenceKind As InstanceReferenceKind = If(boundMeReference.WasCompilerGenerated, InstanceReferenceKind.Implicit, InstanceReferenceKind.Explicit)
+            Dim instanceReferenceKind As InstanceReferenceKind = If(boundMeReference.WasCompilerGenerated, instanceReferenceKind.Implicit, instanceReferenceKind.Explicit)
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >instanceReferenceKind [](start = 107, length = 21)

This looks like an unintentional change in casing.

Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >instanceReferenceKind [](start = 139, length = 21)

And here.

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 4]
```diff
@@ -1073,7 +1073,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
 
         Private Function CreateBoundGotoStatementOperation(boundGotoStatement As BoundGotoStatement) As IBranchStatement
             Dim target As ILabelSymbol = boundGotoStatement.Label
-            Dim branchKind As BranchKind = BranchKind.GoTo
+            Dim branchKind As BranchKind = branchKind.GoTo
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >branchKind [](start = 43, length = 10)

And here.

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 5]
```diff
@@ -1083,7 +1083,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
 
         Private Function CreateBoundContinueStatementOperation(boundContinueStatement As BoundContinueStatement) As IBranchStatement
             Dim target As ILabelSymbol = boundContinueStatement.Label
-            Dim branchKind As BranchKind = BranchKind.Continue
+            Dim branchKind As BranchKind = branchKind.Continue
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >branchKind [](start = 43, length = 10)

And here.

File: src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb [hunk 6]
```diff
@@ -1093,7 +1093,7 @@ Namespace Microsoft.CodeAnalysis.Semantics
 
         Private Function CreateBoundExitStatementOperation(boundExitStatement As BoundExitStatement) As IBranchStatement
             Dim target As ILabelSymbol = boundExitStatement.Label
-            Dim branchKind As BranchKind = BranchKind.Break
+            Dim branchKind As BranchKind = branchKind.Break
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb]: >branchKind [](start = 43, length = 10)

And here.

**Comment by AlekseyTs** [comment]
Done with review pass.

**Comment by 333fred** [comment]
@AlekseyTs addressed feedback.

**Review by AlekseyTs — APPROVED** [review]
LGTM

**Pull request merged**